### PR TITLE
Doc 2 round

### PR DESCRIPTION
**Pull request by lrleon**
## Summary by CodeRabbit

* **Documentation**
  * Added guides for Functional Programming, Iterator Filters, and Threads & Concurrency; refreshed README anchors, examples, and BitArray mention; updated documentation groupings and Doxygen input selection.
* **Reliability / Bug Fixes**
  * Hardened timeout-queue lifecycle, input validation, scheduling, cancellation, timing and race handling; destructor and shutdown behavior improved.
* **API**
  * Expanded timeout-queue control, inspection and safety interfaces.
* **Tests**
  * Added extensive timeout-queue regression, edge-case, concurrency, timing, and ordering tests.
* **Chores**
  * Minor docs, grouping tag updates, build/test orchestration and ignore-file tweaks.

### CODE REVIEW

File: timeoutQueue.C [hunk 1]
```diff
@@ -56,8 +58,7 @@ TimeoutQueue::TimeoutQueue() : isShutdown(false)
 
 TimeoutQueue::~TimeoutQueue()
 {
-  if (not isShutdown)
-    EXIT("TimeoutQueue is not shut down");
+  ah_domain_error_if(not isShutdown) << "TimeoutQueue is not shut down";
```
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.C:66]: @codex

File: Tests/timeoutQueue_test.cc [hunk 2]
```diff
@@ -884,6 +887,306 @@ TEST(TimeoutQueueTest, CancelByIdWithCallback)
...
+      TimeoutQueue q;
+      auto* e = new TestEvent(time_from_now_ms(1000));
+      Time bad = read_current_time();
+      bad.tv_nsec = 2000000000; // Invalid: >= 1e9
+      q.schedule_event(bad, e);
+    },
+    "");
+# else
+  GTEST_SKIP();
+# endif
```
[the remark this answers is not in the document]
Comment (lrleon) [reply Tests/timeoutQueue_test.cc:1092]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2737776721)

File: timeoutQueue.H [hunk 3]
```diff
@@ -224,8 +226,9 @@ public:
 
     ~Event() override
     {
-      ah_runtime_error_if(execution_status == In_Queue)
-        << "Attempting to delete an event in queue (" << this << ")";
+      if (execution_status == In_Queue)
+        std::cerr << "Warning: deleting Event " << this
+                  << " that is still In_Queue" << std::endl;
     }
```
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.H:244]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2738917761)

File: timeoutQueue.C [hunk 4]
```diff
@@ -56,8 +57,15 @@ TimeoutQueue::TimeoutQueue() : isShutdown(false)
...
-  if (not isShutdown)
-    EXIT("TimeoutQueue is not shut down");
+  {
+    std::lock_guard<std::mutex> lock(mtx);
+    if (not isShutdown)
+      {
+        std::cerr << "Warning: TimeoutQueue destructor called without prior shutdown(). "
+                  << "Invoking shutdown() automatically." << std::endl;
+        shutdown_locked();
+      }
```
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.C]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2738917777)

File: Tests/timeoutQueue_test.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (lrleon) [reply Tests/timeoutQueue_test.cc:1092]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2737776721)

File: timeoutQueue.C (hunk 1 above)
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.C:66]: I do not see where ah_domain_error_if is called

**Comment by lrleon** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/lrleon/Aleph-w/pull/15#pullrequestreview-3717566180)

**Comment by lrleon** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/lrleon/Aleph-w/pull/15#pullrequestreview-3717566180)

File: timeoutQueue.C [hunk 5]
```diff
@@ -201,137 +269,152 @@ void TimeoutQueue::triggerEvent()
...
+
+          lock.unlock();
+
+          if (final_status == Event::Deleted)
+            {
+              delete event_to_execute;
               if (callback)
-                callback(event_to_execute, final_status);
+                callback(nullptr, final_status);
             }
```
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.C:337]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2751925545)

File: timeoutQueue.C [hunk 6]
```diff
@@ -201,137 +275,171 @@ void TimeoutQueue::triggerEvent()
...
+                }
               event_to_execute->set_execution_status(Event::Executed);
+            }
+
+          const bool became_empty = prio_queue.size() == 0;
+
+          lock.unlock();
+
+          if (final_status == Event::Deleted)
+            {
```
[the remark this answers is not in the document]
Comment (lrleon) [reply timeoutQueue.C:333]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/15#discussion_r2751956478)
